### PR TITLE
Add remove service

### PR DESCRIPTION
**Pull request by jarrydlee**
cc/ @kwm4385

### CODE REVIEW

**Comment by kalebdavis** [comment]
This is only intended to remove the admin responsibilities from the user, and not remove the user from the group entirely?

File: server/routes/group.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ router.route('/:groupName/user/:userId')
   .delete(validate(paramValidation.userById), groupMiddleware.isGroupAdmin, groupCtrl.removeUser);
 
 router.route('/:groupName/user/:userId/admin')
-  .post(validate(paramValidation.userById), groupMiddleware.isGroupAdmin, groupCtrl.addAdmin);
+  .post(validate(paramValidation.userById), groupMiddleware.isGroupAdmin, groupCtrl.addAdmin)
+  .delete(validate(paramValidation.userById), groupMiddleware.isGroupAdmin, groupCtrl.removeAdmin);
```
Comment (kalebdavis) [inline server/routes/group.js]: Should the user be able to remove themselves from admin?

File: server/services/group.js [hunk 2]
```diff
@@ -115,6 +115,12 @@ function addAdmin(groupName, userId) {
     .then(validatedUserId => Group.addAdmin(groupName, validatedUserId));
 }
 
+function removeAdmin(groupName, userId) {
+  return userService.exists(userId)
```
Comment (kalebdavis) [inline server/services/group.js]: Should there be some kind of logic to ensure that the group isn't left without an admin? Ie, you can't remove an admin if it'll leave the array empty?

File: server/routes/group.js (hunk 1 above)
Comment (jarrydlee) [inline server/routes/group.js]: This is added in the service

File: server/services/group.js [hunk 3]
```diff
@@ -19,6 +21,17 @@ function deleteGroup(groupName) {
   return Group.delete(groupName);
 }
 
+function canRemoveAdmin(groupName) {
```
Comment (kalebdavis) [inline server/services/group.js]: The naming here implies that the method will return a boolean, when it's actually returning a group or an error. Maybe `getGroupToRemoveAdmin` or something instead?

Comment (jarrydlee) [inline server/services/group.js]: I think I'd rather just have it not return anything and just do the check and throw if there is a sole admin

Comment (kalebdavis) [inline server/services/group.js]: That works for me as well

File: server/models/group.js [hunk 4]
```diff
@@ -26,11 +26,18 @@ const GroupSchema = new mongoose.Schema({
...
-    type: mongoose.Schema.Types.ObjectId,
-    ref: 'User',
-    default: []
-  }],
+  admins: {
+    type: [{
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'User',
+    }],
+    validate: {
```
Comment (kalebdavis) [inline server/models/group.js]: 👍

**Pull request merged**